docs-src/doctype-starter-doc.md

```diff
@@ -0,0 +1,362 @@
+# Create a Ceramic Doctype
+
+In order to implement a new Ceramic Doctype there are a couple of interfaces which should be followed:
+
+- **Doctype**
+	- This interface is used to implement an actual document.
+- **DoctypeHandler**
+	- This interface is used to implement the state transition handler for the document which will be executed by the Ceramic node.
+
+There are other constructs which need to be used in order to create a `Doctype`:
+- **DoctypeStatic**
+	- Decorator for the Doctype implementation
+- **DoctypeConstructor**
+	- Describes the Doctype constructor which needs to be implemented.
+
+One interface worth noting is the `CeramicAPI` interface which lets the developer communicate with the Ceramic node in two ways:
+- **directly**: the node is the process running on the same machine
+- **indirectly**: the node is running somewhere else and the communication is done over the HTTP protocol.
+	
+# Doctype
+The interface is listed in the code snipped below.
+
+```typescript
+/**  
+ * Describes common doctype attributes 
+ */
+ abstract class Doctype extends EventEmitter {  
+    constructor(private _state: DocState, private _context: Context) {  
+        super()  
+    }  
+  
+    get id(): string {  
+        return DoctypeUtils.createDocId(this.state.log[0])  
+    }  
+  
+    get doctype(): string {  
```

Comment:
Unrelated to this CR, but I really think we need to either rename the Doctype class or rename every place we refer to the string name of a doctype.  I'd love to rename the class but I can't actually think of a better name for it.  So maybe the thing to do is change everyplace where we're referring to the type name to be "doctypename" (doctypeName? doctype_name?)

Anyway this is totally an aside, just something that's been bugging me

Reply:
Not sure we need this property anymore since you can get it from the DocID?